newIDE/app/src/GameDashboard/Feedbacks/FeedbackCard.js

```diff
@@ -0,0 +1,170 @@
+// @flow
+
+import * as React from 'react';
+import { t, Trans } from '@lingui/macro';
+import { I18n } from '@lingui/react';
+import { type I18n as I18nType } from '@lingui/core';
+
+import { Card } from '@material-ui/core';
+import CheckCircleOutlineIcon from '@material-ui/icons/CheckCircleOutline';
+import CheckCircleIcon from '@material-ui/icons//CheckCircle';
+
+import { ResponsiveLineStackLayout } from '../../UI/Layout';
+import CardContent from '../../UI/Card/CardContent';
+import Text from '../../UI/Text';
+import { Column, LargeSpacer, Line, Spacer } from '../../UI/Grid';
+import IconButton from '../../UI/IconButton';
+import GDevelopThemeContext from '../../UI/Theme/ThemeContext';
+
+import Rating from './Rating';
+
+import {
+  updateComment,
+  type Comment,
+  type GameRatings,
+} from '../../Utils/GDevelopServices/Play';
+import { useDebounce } from '../../Utils/UseDebounce';
+import { type AuthenticatedUser } from '../../Profile/AuthenticatedUserContext';
+import { showErrorBox } from '../../UI/Messages/MessageBox';
+
+const styles = { textComment: { whiteSpace: 'pre-wrap' } };
+
+type Props = {|
+  comment: Comment,
+  authenticatedUser: AuthenticatedUser,
+|};
+
+const getRatings = (ratings: ?GameRatings) => {
+  if (!ratings) return null;
+  if (ratings.version === 1) {
+    return [
+      {
+        key: 'rating-sound',
+        label: <Trans>Sound</Trans>,
+        value: ratings.sound,
+      },
+      {
+        key: 'rating-visuals',
+        label: <Trans>Visuals</Trans>,
+        value: ratings.visuals,
+      },
+      { key: 'rating-fun', label: <Trans>Fun</Trans>, value: ratings.fun },
+      {
+        key: 'rating-ease-of-use',
+        label: <Trans>Ease of use</Trans>,
+        value: ratings.easeOfUse,
+      },
+    ];
+  }
+};
+
+const getCurrentState = (pendingState: ?boolean, state: boolean) => {
+  return pendingState === null ? state : pendingState;
+};
+
+const FeedbackCard = ({ comment, authenticatedUser }: Props) => {
+  const { getAuthorizationHeader, profile } = authenticatedUser;
+  const ratings = getRatings(comment.ratings);
+  const theme = React.useContext(GDevelopThemeContext);
+
+  const [processed, setProcessed] = React.useState<boolean>(
+    !!comment.processedAt
+  );
+  const [
+    pendingProcessedState,
+    setPendingProcessedState,
+  ] = React.useState<?boolean>(null);
```

Comment:
Can we do everything with only 1 value in the state?

(not sure, with the fact that a user can click twice, and only the second call would be registered, and if fails, then we need to revert to the previous "saved" value)

Reply:
I don't think we can; with only one state, we can't know what was the first state while the call is debounced.
As you said, if I click two times, it will do "true -> false -> true" the call will be sent at the second true and I don't think we can easily get back to the first "true" if the call fails

Reply:
I managed to abstract this logic in a useOptimisticState so that it will be easier to read.